cli/azd/pkg/config/manager.go

```diff
@@ -106,10 +107,31 @@ func GetUserConfigDir() (string, error) {
 			return "", fmt.Errorf("could not determine current home directory: %w", err)
 		}
 
-		configDirPath = filepath.Join(homeDir, configDir)
+		configDirPath = filepath.Join(homeDir, cConfigDir)
+	}
+
+	err := os.MkdirAll(configDirPath, osutil.PermissionDirectoryOwnerOnly)
+	if err != nil {
+		return configDirPath, err
+	}
+
+	// Ensure that the "x" permission is set on the folder for the current
+	// user. In cases where the config directory is ~/.azd, OS upgrades and
+	// other processes can remove the "x" permission
+	if runtime.GOOS == "linux" || runtime.GOOS == "darwin" {
+		info, err := os.Stat(configDirPath)
+		if err != nil {
+			return configDirPath, err
+		}
+
+		permissions := info.Mode().Perm()
+		if permissions&0100 == 0 {
+			// Ensure user execute permissions
+			err := os.Chmod(configDirPath, permissions|100)
```

Comment:
```suggestion
			err := os.Chmod(configDirPath, permissions|0100)
```

Reply:
We might just want to create a constant of this mask, given that it's easy for us to forget the leading 0 and only some of the time the linter catches us and tells us that we were likely doing something wrong.